src/tools/ScenarioMeasurement/Startup/ProfileParser.cs

```diff
@@ -13,19 +13,21 @@ internal class ProfileParser : IParser
         IParser other;
         public ProfileParser(IParser other) => this.other = other;
 
-        public void EnableKernelProvider(TraceEventSession kernel)
+        public void EnableKernelProvider(ITraceSession kernel)
         {
-            kernel.StackCompression = true;
+            TraceEventSession kernelSession = ((WindowsTraceSession)kernel).KernelSession;
+            kernelSession.StackCompression = true;
             var keywords = KernelTraceEventParser.Keywords.Default;
-            kernel.EnableKernelProvider(keywords, keywords);
+            kernelSession.EnableKernelProvider(keywords, keywords);
         }
 
-        public void EnableUserProviders(TraceEventSession user)
+        public void EnableUserProviders(ITraceSession user)
         {
-            // make sure we turn on whatever the user wanted so the start/stops are findable.
             other.EnableUserProviders(user);
-            user.EnableProvider(ClrTraceEventParser.ProviderGuid, Microsoft.Diagnostics.Tracing.TraceEventLevel.Verbose, (ulong)ClrTraceEventParser.Keywords.Default);
-            user.EnableProvider(ClrPrivateTraceEventParser.ProviderGuid, Microsoft.Diagnostics.Tracing.TraceEventLevel.Verbose, (ulong)(ClrPrivateTraceEventParser.Keywords.GC
+            TraceEventSession userSession = ((WindowsTraceSession)user).UserSession;
```

Comment:
Does this work on Linux? I don't see where the profile iteration gets skipped or anything.

Reply:
It is skipped in ```TraceSessionManager.CreateProfileSession()```. It might be safer if we check the platform again here, but I'm not sure if an ```IParser``` should know that.